test/featurizations/GraphNodeFeaturization_tests.jl

```diff
@@ -9,33 +9,40 @@ using ChemistryFeaturization.Featurization
     # custom lookup table...
     test_df = CSV.read(abspath(@__DIR__, "..", "test_data", "lookup_table.csv"), DataFrame)
 
-    @testset "Construct, encode, decode" begin
+    local fnames = ["X", "Block", "Atomic mass"]
+
+    @testset "Encode" begin
         # make sure both constructors give the same results
-        local fnames = ["X", "Block", "Atomic mass"]
         features = ElementFeatureDescriptor.(fnames)
         fzn1 = GraphNodeFeaturization(features)
         fzn2 = GraphNodeFeaturization(fnames)
 
         triangle_C_1 = AtomGraph(Float32.([0 1 1; 1 0 1; 1 1 0]), ["C", "C", "C"])
         triangle_C_2 = AtomGraph(Float32.([0 1 1; 1 0 1; 1 1 0]), ["C", "C", "C"])
-        # (this one also tests broadcasting of featurize!)
-        featurize!.([triangle_C_1, triangle_C_2], [fzn1, fzn2])
-        @test triangle_C_1.encoded_features == triangle_C_2.encoded_features
+        # @test fzn1 == fzn2
+
+        featurized_1, featurized_2 = featurize.([triangle_C_1, triangle_C_2], [fzn1, fzn2]) # featurize can be broadcasted
+        @test featurized_1 == featurized_2
+    end
 
+    @testset "Decode" begin
         # test that other options work properly
-        fzn3 = GraphNodeFeaturization(fnames, nbins = 2)
-        F2 = AtomGraph(Float32.([0 1; 1 0]), ["F", "F"])
-        featurize!(F2, fzn3)
-        decoded_matrix = decode(fzn3, F2.encoded_features)
-        decoded_ag = decode(F2)
-        enc1 = F2.encoded_features
-        @test all(
-            map(d -> d[1]["Block"] == d[2]["Block"] == "p", [decoded_matrix, decoded_ag]),
-        )
-        fzn4 = GraphNodeFeaturization(fnames, nbins = [2, 4, 2])
-        F2 = AtomGraph(Float32.([0 1; 1 0]), ["F", "F"])
-        featurize!(F2, fzn4)
-        @test all(F2.encoded_features .== enc1)
+
+        # TODO - Figure out a way to robustly test this. Would creating a custom FD type whose encoded/decoded value you know beforehand and directly compare with be the best thing to do here?
```

Comment:
I'm not sure I understand why what was there before wouldn't work?

Reply:
Only `Block` seems to be tested. Wouldn't it be more robust if all features defined in whatever `fnames` is passed to `GNF` could somehow be tests?

FWIW, we might need to think of a cleaner way to structure these tests too, because these tests for `GraphNodeFeaturization` are very similar to those the new container type (`FeaturizedAtoms`) would require.

Reply:
yeah it would probably make sense to test more features, and also agreed about restructuring the tests. I'll leave it up to you if you want to do that before merging or just file an issue to remember to do it later

Reply:
Add it to the Project board, and we'll get to it? I think it might be a better idea to work on those changes in a different PR.